src/main/resources/com/puppycrawl/tools/checkstyle/grammars/javadoc/JavadocLexer.g4

```diff
@@ -120,7 +120,8 @@ Newline5: NEWLINE
 Leading_asterisk3: LEADING_ASTERISK -> type(LEADING_ASTERISK);
 XmlTagOpen1: '<' -> type(OPEN), pushMode(xmlTagDefinition);
 STRING: '"' .*? '"' {referenceCatched = false;} -> mode(DEFAULT_MODE);
-PACKAGE: [a-z_$] ([a-z0-9_$] | '.')+ [a-z0-9_$] {referenceCatched = true;};
+PACKAGE: [a-z_$] ([a-z0-9_$] | '.')+ [a-z0-9_$] {_input.LA(1) == '.' || _input.LA(1) == '}'
+                     || Character.isWhitespace(_input.LA(1))}? {referenceCatched = true;};
```

Comment:
@romani I didn't understand why did you insert `WS` _(whitespace)_ in package token and how is that acceptable for a valid package name to have `WS`. Can you please explain it to me ?